### ISSUE TITLE
ximgproc: warnings during build samples on CLANG 

### ISSUE THREAD

**Issue by seiko2plus** [opencv/opencv_contrib#1916]
##### System information (version)

- OpenCV => 3.4, master
- Operating System / Platform => current test ppc64le GNU/Linux, but it should be other architectures too ,, gonna update it later
- Compiler => clang 5, 6

##### Detailed description

````
worker/buildbot/Power8_Linux_clang-5__opencv/opencv_contrib/modules/ximgproc/samples/colorize.cpp:341:56: warning: implicit conversion from 'double' to 'int' changes value from 0.1 to 0 [-Wliteral-conversion]
/worker/buildbot/Power8_Linux_clang-5__opencv/opencv_contrib/modules/ximgproc/samples/colorize.cpp:345:56: warning: implicit conversion from 'double' to 'int' changes value from 0.1 to 0 [-Wliteral-conversion]

````
CI URLS:
- CLANG 5: https://ocv-power.imavr.com/#/builders/8/builds/1
- CLANG 6: https://ocv-power.imavr.com/#/builders/15/builds/1
##### Steps to reproduce
build opencv on clang

**Comment by utkarshg6** [comment]
Hey, @alalek . I would like to work on this issue. Which CLANG version should I use?

**Comment by seiko2plus** [comment]
@utibenkei, @alalek, this issue already resolved by #2011 for 3.4 branch and #1980 for master branch.
